platform/security/idp/security-idp-plugin/security-idp-plugin-subjectconfirmation/src/main/java/org/codice/ddf/security/idp/plugin/subjectconfirmation/SubjectConfirmationPresignPlugin.java

```diff
@@ -88,27 +87,11 @@ private Audience buildAudience(AudienceBuilder audienceBuilder, String uri) {
     return audience;
   }
 
-  private void addAudiences(
-      AuthnRequest authnRequest,
-      Response response,
-      List<String> spMetadata,
-      Set<Binding> supportedBindings) {
+  private void addAudiences(AuthnRequest authnRequest, Response response) {
     AudienceBuilder audienceBuilder = new AudienceBuilder();
     AudienceRestrictionBuilder audienceRestrictionBuilder = new AudienceRestrictionBuilder();
 
-    Collection<EntityInformation> entityInformationCollection =
-        parseServiceProviderMetadata(spMetadata, supportedBindings).values();
-
-    List<Audience> audienceList =
-        entityInformationCollection
-            .stream()
-            .map(
-                ei ->
-                    ei.getAssertionConsumerService(
-                        authnRequest, null, authnRequest.getAssertionConsumerServiceIndex()))
-            .map(ServiceInfo::getUrl)
-            .map(uri -> buildAudience(audienceBuilder, uri))
-            .collect(Collectors.toList());
+    Audience audience = buildAudience(audienceBuilder, authnRequest.getIssuer().getValue());
```

Comment:
✏️ The overloading of elements in the SAML spec can lead to some confusion. E.g. I was going to comment here that the `ISSUER` of the assertion is the IdP, not the SP...until I realized you were pulling the `ISSUER` from the `authnRequest`. Maybe a short comment here to quote the spec, something like...
> On an AuthnRequest, "[t]he `<Issuer>` element MUST be present and MUST contain the unique identifier of the requesting service provider...

Reply:
Yeah, you're right. I'll throw a comment in there.